docker-compose.yml

```diff
@@ -0,0 +1,47 @@
+version: '3.9'
+
+
+
+services:
+  ficai-db:
+    container_name: ficai-db
+    image: postgres:14-alpine
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges:true
+    expose:
+      - 5432
+    environment:
+      POSTGRES_DB: $FICAI_DB_DATABASE
+      POSTGRES_USER: $FICAI_DB_USERNAME
+      POSTGRES_PASSWORD: $FICAI_DB_PASSWORD
+    volumes:
+      - './schema.sql:/docker-entrypoint-initdb.d/schema.sql'
+      - 'db_data:/var/lib/postgresql/data/'
+    healthcheck:
+      test: [ "CMD-SHELL", "pg_isready -U $FICAI_DB_USERNAME -d $FICAI_DB_DATABASE" ]
+      interval: 10s
+      timeout: 5s
+      retries: 10
+      start_period: 10s
+
+  ficai-signals:
+    container_name: ficai-signals
+    build: .
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges:true
+    depends_on:
+      - ficai-db
+    ports:
+      - '8080:8080'
+    env_file:
+      - .env
```

Comment:
I'm in favor of hardcoding everything besides password. Which, ideally, should be stored in separate plain file to make use of docker secrets if it's gonna run anywhere near production. Though it might make initial setup a bit trickier for users.

Reply:
That's probably fine, but I don't completely follow the logic. That would prevent someone who can directly access the docker-compose environment but _not_ the swarm secrets from doing something malicious in the db? What exactly is the threat model here?

We should expand the README to list what is actually expected to be in `.env` (and maybe rename it `docker-compose.env`?) for use in docker-compose, and give recommendations on setting them -- maybe reuse the `FICAI_PWD_PEPPER` instructions for `FICAI_DB_PASSWORD`?